### PR TITLE
Cleanup redundant parameters

### PR DESCRIPTION
**Pull request by sregales-TNO**
**Approach**
Remove and replace redundant parameters.

### CODE REVIEW

File: libres/lib/include/ert/enkf/enkf_main.hpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ bool enkf_main_smoother_update(enkf_main_type *enkf_main,
 void enkf_main_create_run_path(enkf_main_type *enkf_main,
                                const ert_run_context_type *run_context);
 
-enkf_main_type *enkf_main_alloc(const res_config_type *, bool, bool);
+enkf_main_type *enkf_main_alloc(const res_config_type *, bool = false);
```
Comment (dafeda) [inline libres/lib/include/ert/enkf/enkf_main.hpp:82]: Do we want to set default to false both in header and cpp?

Comment (dafeda) [inline libres/lib/include/ert/enkf/enkf_main.hpp:82]: Do we want to set default to false both in header and cpp?

Comment (sregales-TNO) [reply libres/lib/include/ert/enkf/enkf_main.hpp:82]: I have realize that most calls to the function always have it as false. So I thought to keep it false until explicitly told to be true. I know that is always better to be explicit than implicit, but what is the value of the if almost all calls are `false`

Comment (sregales-TNO) [reply libres/lib/include/ert/enkf/enkf_main.hpp:82]: plus is not set as defualt on the exposed endpoint on python

**Review by dafeda — APPROVED** [review]
LGTM 👍

**Pull request merged**